### DISCUSSION TITLE
random strings are not compressed

### DISCUSSION THREAD

**Discussion by lxzan** [klauspost/compress#827] [Q&A] [answered]
I found that 1000 byte random strings are not compressed when I use flate compression, which is inconsistent with the behavior of the standard library. Does it meet your normal expectations?

The link is to the random string generation algorithm:
https://github.com/lxzan/gws/blob/master/internal/random.go

**Comment by klauspost** [comment, marked as answer]
There is no point in compressing them, if they cannot be compressed. 

It will just create (potentially) longer output and be much slower to decompress.

The compressor will estimate if there are gains to compressing. If there is no or only a very small gain it will be stored uncompressed.

**Reply by lxzan** [reply]
This is smart! Thanks for answering questions.

**Reply by klauspost** [reply]
This is the function that does it: https://github.com/klauspost/compress/blob/master/flate/huffman_bit_writer.go#L618

It chooses between:

* Uncompressed
* Predefined table
* Reuse table from last block
* Create new table for block
